components/ee/payment-endpoint/src/accounting/subscription-service.ts

```diff
@@ -101,7 +101,7 @@ export class SubscriptionService {
     async hasActiveAndNotYetCancelledPaidSubscription(userId: string, date: Date): Promise<boolean> {
         const subscriptions = await this.accountingDB.findActiveSubscriptionsForUser(userId, date.toISOString());
         return subscriptions
-            .filter(s => Subscription.isActive(s, date.toISOString()) && !Subscription.isCancelled(s))
+            .filter(s => Subscription.isActive(s, date.toISOString()))
```

Comment:
The method does what it says on the tin (name), and is used in the other places correctly I think.

But [this call](https://github.com/gitpod-io/gitpod/blob/e8762f258e8b81d0a8da0b8062e7352722948e6c/components/server/ee/src/user/eligibility-service.ts#L255) is odd indeed: I agree that we should call `hasActivePaidSubscription` with the implementation you suggested.

Reply:
@jankeromnes I faintly recall we added made the change here together, right? Do you recall what the motivation was for checking `isActive && !isCancelled` instead of `isActive`? IMO this was because of a user that:
 1. bought a subscription
 1. quickly cancelled it (and we reimbursed them)
 1. expected to be back in trial mode immediately.
 
In general the user-frriendly pattern of "easy-in-easy-out" makes sense here.